src/cpp/src/cache_manager.hpp

```diff
@@ -56,8 +59,22 @@ class CacheManager {
             for (size_t decoder_layer_id = 0; decoder_layer_id < m_device_config.get_num_layers(); ++decoder_layer_id) {
                 ov::Shape value_cache_shape = set_first_dim_and_make_static(m_device_config.get_value_cache_shape(decoder_layer_id), num_kv_blocks);
                 ov::Shape key_cache_shape = set_first_dim_and_make_static(m_device_config.get_key_cache_shape(decoder_layer_id), num_kv_blocks);
+#ifdef _WIN32
                 ov::Tensor key_cache(m_device_config.get_cache_precision(), key_cache_shape);
                 ov::Tensor value_cache(m_device_config.get_cache_precision(), value_cache_shape);
+#else
+                auto key_size = ov::shape_size(key_cache_shape) * m_device_config.get_cache_precision().size();
+                auto value_size = ov::shape_size(value_cache_shape) * m_device_config.get_cache_precision().size();
+
+                // allocate memory filled with zeros
+                auto key_ptr = mmap(NULL, key_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+                auto value_ptr = mmap(NULL,  value_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+                OPENVINO_ASSERT(key_ptr != MAP_FAILED);
+                OPENVINO_ASSERT(value_ptr != MAP_FAILED);
+                ov::Tensor key_cache = ov::Tensor(m_device_config.get_cache_precision(), key_cache_shape, key_ptr);
+                ov::Tensor value_cache = ov::Tensor(m_device_config.get_cache_precision(), value_cache_shape, value_ptr);
```

Comment:
how this memory is de-allocated? should we call `unmap`?

`ov::Tensor` does not hold the memory passed to it and custom allocator is required, example is https://github.com/openvinotoolkit/openvino.genai/blob/ef1f0f60c58cf0452cfbc6305f5cf30e9137ceb7/src/python/py_image_generation_pipelines.cpp#L118-L143

Reply:
> how this memory is de-allocated? should we call `unmap`?

According to this docs https://man7.org/linux/man-pages/man2/mmap.2.html the region is automatically unmapped when the process is terminated. If I understood it correctly explicit `unmap` is needed mainly for files. 
But as we need custom allocator, so I added `unmap` there anyway, to be sure.